hieroweb/web/src/hiero/web/InitialObject.java

```diff
@@ -2,34 +2,36 @@
 
 import org.checkerframework.checker.nullness.qual.NonNull;
 import org.hiero.sketch.dataset.LocalDataSet;
-import org.hiero.sketch.dataset.api.IJson;
+import org.hiero.sketch.dataset.ParallelDataSet;
+import org.hiero.sketch.dataset.api.IDataSet;
 import org.hiero.sketch.table.Table;
 import org.hiero.sketch.table.api.ITable;
 
 import javax.websocket.Session;
+import java.util.ArrayList;
 
 public class InitialObject extends RpcTarget {
-    // Used as a reply for loadTable.
-    static class RemoteTableId implements IJson {
-        public final String id;
-        RemoteTableId(String id) { this.id = id; }
-    }
-
-    public InitialObject(@NonNull String objectId) {
-        super(objectId);
-    }
-
     @HieroRpc
     void loadTable(@NonNull RpcRequest request, @NonNull Session session) {
         // TODO: look at request.  Now we just supply always the same table
-        String id = this.server.freshId();
         Table t = Table.testTable();
-        LocalDataSet<ITable> data = new LocalDataSet<ITable>(t);
-        TableTarget table = new TableTarget(id, data);
-        this.server.addObject(table);
+        final int parts = 5;
+        ArrayList<IDataSet<ITable>> fragments = new ArrayList<IDataSet<ITable>>();
+        for (int i = 0; i < parts; i++) {
+            LocalDataSet<ITable> data = new LocalDataSet<ITable>(t);
+            fragments.add(data);
+        }
+        IDataSet<ITable> big = new ParallelDataSet<ITable>(fragments);
+        TableTarget table = new TableTarget(big);
+        RpcObjectManager.instance.addObject(table);
 
-        RpcReply reply = request.createReply(new RemoteTableId(id));
-        this.server.sendReply(reply, session);
+        RpcReply reply = request.createReply(table.idToJson());
```

Comment:
Unfortunately this won't work, because a method may return multiple results. The server has no idea when the method is finished. So each @HieroRpc method is responsible for returning all replies and closing the session, which implies "end of replies".

It's even worse in general, because the replies may be generated on a separate thread. I will have to document this.